### PR TITLE
Init proposal for new service framework

### PR DESCRIPTION
**Pull request by loicsaintroch**
# Changes proposed ✍️

### <samp>🤖 Generated by Copilot at f12576c</samp>

This pull request adds two new services to the scaffold: `template_consumer` and `template_worker`. These services are examples of how to use `kafka` and `temporal` for message streaming and workflow orchestration. They are written in TypeScript and use common packages and configurations from the `@crowd` namespace. They also have ESLint and Prettier settings for code quality and consistency.
​

### <samp>🤖 Generated by Copilot at f12576c</samp>

> _New services added_
> _`kafka` and `temporal`_
> _Cutting-edge scaffold_

### <samp>🤖 Generated by Copilot at f12576c</samp>

*  Add kafka and temporal services to the scaffold.yaml file ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-77a0fef8de0440f0d77af31bfe59b95174e87032eea32824f88332487501e787R138-R162))
*  Create a Dockerfile for the temporal service with curl, temporal CLI, and exposed ports ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-7f8cbbd355f6c4cd5656d2d8adc45f48ab5c8d1804356887e9b0dfa97dc27c4aR1-R10))
*  Create template_consumer and template_worker services with TypeScript, ESLint, and Prettier configurations ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-3d432e468a399024e98bcf3016a735b6f55b337dd45ebe808e5652440c559ee5R1-R21), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-557e21d4841ce0ceccef4c79e1a35a5e8aff8a845720817469181433976c0846R1-R3), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-942bb4552e1d922f4e103a89d4901277a3c2ea14f009d78e2eb776fb1b66ec5bR1-R7), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-7551e33fe0694e1e185a0d1fc0598c72d95c17ab6cc2ac7cf25e83307bab475eR1-R35), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-737ff82050e39f7ab7b046201df2b83316410388a06885026d9f565438628fddR1-R17), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-358ef79115654abcae5564dcf9a42ec7e8ee759c57ed14f97e78e9214f82109fR1-R21), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-979452a86f881567f6a3871cd0edc52bc75b470ab61a98564a7f6a1d07fb8de2R1-R3), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-b05115cc47e5f29fd29777ae95843fdc7bd94976de3fce33e671ba8102f87366R1-R7), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-386d778609708470d9cfd81b6bfbb7f8dd86c7e23cd795e640d090703bb6d0d5R1-R36), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-f386351bcf80c72390095bc4a86f5dd775a969b2f325105a8ae6a466a4c9c7feR1-R17))
*  Implement ServiceConsumer and ServiceWorker classes in the main.ts files of the template services, using the @crowd packages ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-e904cf1d6e8e1fa30fbab3e76c9103dd53776f871f2cdf8b7f88f47c2227877cR1-R34), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-088464fd4198c4ed373bac8ccdc1158aa3fa36947cdd1e4fb82440f89c9a41fdR1-R26))
*  Define an example greet activity and workflow in the template_worker service, using the @temporalio package ([link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-b2773f79f884147f6a5362427e27b895f2ed720f74ef4f585e38237f35e04132R1-R3), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-a8a74ab1ead57c03e58d3a106e14a51aac061e5ee7d7292de93934506a41a4a7R1-R3), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-7253575b652ecb46d88b3ee30149c2957248ad70f6a92065f99792042ea4b026R1-R3), [link](https://github.com/CrowdDotDev/crowd.dev/pull/1697/files?diff=unified&w=0#diff-59848c4f32f1f2b972ebe021486c7f69f38e5b61d2ba9b53938725fbc45d9b5aR1-R16))

## Checklist ✅
- [ ] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screehshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [ ] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

### CODE REVIEW

File: scripts/scaffold/temporal/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM alpine:latest
+
+RUN apk add curl
+RUN curl -sSf https://temporal.download/cli.sh | sh
```
Comment (sausage-todd) [inline scripts/scaffold/temporal/Dockerfile:4]: Why not https://hub.docker.com/r/temporalio/server ?

Comment (loicsaintroch) [reply scripts/scaffold/temporal/Dockerfile:4]: The server image needs to be configured for development purpose. Also, it's heavier than just downloading the CLI for running a complete stack in development mode, with UI enabled and SQLite.

File: services/apps/template_consumer/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "@types/node": "^20.8.2",
+    "config": "^3.3.9",
+    "ts-node": "^10.9.1",
+    "tsconfig-paths": "^4.2.0",
+    "typescript": "^5.2.2"
+  },
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^6.7.4",
+    "@typescript-eslint/parser": "^6.7.4",
+    "concurrently": "^8.2.1",
```
Comment (themarolt) [inline services/apps/template_consumer/package.json:28]: I don't think we need concurrently anymore

File: services/apps/template_consumer/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    retries: 1000,
+  },
+}
+
+const svc = new ServiceConsumer(config, options)
+
+setImmediate(async () => {
+  await svc.start()
+
+  // await svc.consumer.run({
```
Comment (themarolt) [inline services/apps/template_consumer/src/main.ts:31]: is this required? why is this commented out?

File: services/apps/template_consumer/src/main.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    enabled: false,
+  },
+}
+
+const options: Options = {
+  maxWaitTimeInMs: 1000,
+  retryPolicy: {
+    initialRetryTime: 1000,
+    maxRetryTime: 1000,
+    retries: 1000,
```
Comment (themarolt) [inline services/apps/template_consumer/src/main.ts:22]: should this be configurable per service? maybe we need env variables for these. also retries being at 1000 feels like too much :D wouldn't this hinder processing power if activity/workflow has logical errors?

**Review by themarolt — CHANGES_REQUESTED** [review]

File: services/apps/template_consumer/src/main.ts (hunk 3 above)
Comment (loicsaintroch) [reply services/apps/template_consumer/src/main.ts:31]: This is a template/example/demo service just to show it works. If not commented, linter and CI fails.

Comment (themarolt) [reply services/apps/template_consumer/src/main.ts:31]: so in a real consumer/worker this is required?

**Review by themarolt — APPROVED** [review]

**Pull request merged**